src/lv_core/lv_obj.c

```diff
@@ -1592,6 +1592,25 @@ void lv_obj_get_coords(const lv_obj_t * obj, lv_area_t * cords_p)
     lv_area_copy(cords_p, &obj->coords);
 }
 
+/**
+ * Adjust the coordinates retrieved from lv_obj_get_coords() according
+ * to the object's style.
+ */
+void lv_obj_adjust_coords(const lv_obj_t *obj, lv_area_t * cords_p)
+{
+	const lv_style_t *style = lv_obj_get_style(obj);
+	if(style->body.border.part & LV_BORDER_LEFT)
+		cords_p->x1 += style->body.border.width;
+
+	if(style->body.border.part & LV_BORDER_RIGHT)
+		cords_p->x2 -= style->body.border.width;
+	if(style->body.border.part & LV_BORDER_TOP)
+		cords_p->y1 += style->body.border.width;
+	if(style->body.border.part & LV_BORDER_BOTTOM)
+		cords_p->y2 -= style->body.border.width;
```

Comment:
It's a little bit dangerous to change the coordinates this way. 
After a coordinate change `LV_SIGNAL_COORD_CHG` should be sent to notify the object about the new coordinates. However, if the object has some `fit` feature enabled, it will revert the coordinates in the signal function.

What about updating `lv_roller_set_visible_row_count` to take border into account?